agnpy/spectra/tests/test_spectra.py

```diff
@@ -803,11 +810,14 @@ def test_compare_calculation_with_calculation_split_into_two(self):
         synch = Synchrotron(blob)
 
         # iterate over 60 s in 20 steps
-        eval_1 = initial_n_e.evaluate_time(60*u.s, synch.electron_energy_loss_per_time, subintervals_count=20)
+        blob.n_e_time_evolution(synch.electron_energy_loss_rate, 60 * u.s, subintervals_count=20)
+        eval_1 = deepcopy(blob.n_e)
         # iterate first over 30 s, and then, starting with interpolated distribution, over the remaining 30 s,
         # with slightly different number of subintervals
-        eval_2 = initial_n_e.evaluate_time(30 * u.s, synch.electron_energy_loss_per_time, subintervals_count=10)\
-            .evaluate_time(30 * u.s, synch.electron_energy_loss_per_time, subintervals_count=8)
+        blob.n_e = initial_n_e
+        blob.n_e_time_evolution(synch.electron_energy_loss_rate, 30 * u.s, subintervals_count=10)
+        blob.n_e_time_evolution(synch.electron_energy_loss_rate, 30 * u.s, subintervals_count=8)
+        eval_2 = deepcopy(blob.n_e)
```

Comment:
here in principle you do not need deepcopy, because eval_2 is not being modified anymore